### PR TITLE
made solution

### PR DESCRIPTION
**Pull request by Sashk52**

### CODE REVIEW

File: src/main/java/taxi/dao/CarDao.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    Car create(Car car);
+
+    Optional<Car> get(Long id);
+
+    List<Car> getAll();
+
+    Car update(Car car);
+
+    boolean delete(Long id);
+}
```
Comment (mfedenko09) [inline src/main/java/taxi/dao/CarDao.java:19]: create this method please List<Car> getAllByDriver(Long driverId);

File: src/main/java/taxi/service/CarServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        car.getDrivers().remove(driver);
+        carDao.update(car);
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
```
Comment (mfedenko09) [inline src/main/java/taxi/service/CarServiceImpl.java]: this implementation should be on dao level
```suggestion
    public List<Car> getAllByDriver(Long driverId) {
```

**Review by mfedenko09 — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/dao/CarDao.java (hunk 1 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/CarDao.java:19]: done

File: src/main/java/taxi/service/CarServiceImpl.java (hunk 2 above)
Comment (Sashk52) [reply src/main/java/taxi/service/CarServiceImpl.java]: done

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/taxi/service/CarServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        List<Driver> newDriverList = car.getDrivers();
+        newDriverList.add(driver);
+        car.setDrivers(newDriverList);
```
Comment (RomanDubovskyi) [inline src/main/java/taxi/service/CarServiceImpl.java]:
```suggestion
        car.getDrivers().add(driver);
```

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**